openapi-spec/swagger.yaml

```diff
@@ -2424,44 +2489,93 @@ components:
                 lvPath: string
                 nfsFileshareID: string
                 nfsFileshareName: string
-    FileShareSnapshotSpec:
+    FileShareSnapshotCreateSpec:
       description: Snapshot is a description of fileshare snapshot resource.
       allOf:
-        - $ref: '#/components/schemas/BaseModel'
-        - type: object
-          required:
-            - tenantId
+      - type: object
+        required:
             - name
-            - status
-            - snapshotId
             - fileshareId
-          properties:
-            fileshareId:
-              type: string
-            tenantId:
-              type: string
-              readOnly: true
-            userId:
-              type: string
-              readOnly: true
-            name:
-              type: string
-            description:
-              type: string
-            status:
+        properties:
+          fileshareId:
+            type: string
+          name:
+            minLength: 1
+            maxLength: 255
+            type: string
+          description:
+            type: string
+          status:
+            type: string
+            readOnly: true
+          profileId:
+            type: string
+          metadata:
+            type: object
+            readOnly: true
+            additionalProperties:
               type: string
-              readOnly: true
-            profileId:
+            example:
+              snapshotId: string
+              snapshotName: string
+              lvPath: string
+    FileShareSnapshotUpdateSpec:
+      description: Snapshot is a description of fileshare snapshot resource.
+      allOf:
+      - type: object
+        properties:
+          fileshareId:
+            type: string
+          name:
+            minLength: 1
+            maxLength: 255
+            type: string
+          description:
+            type: string
+    FileShareSnapshotRespSpec:
+      description: Snapshot is a description of fileshare snapshot resource.
+      allOf:
+      - $ref: '#/components/schemas/BaseModel'
```

Comment:
Is the reference of Base model correct?

Reply:
it's correct in response, because basemodel just contains id createat and updatedat